docs/design/coreclr/botr/mscorlib.md

```diff
@@ -1,355 +1,356 @@
+Clearly there are places where the CLR must call into managed code from native. For this purpose, we have added a `MethodDescCallSite` class to handle a lot of plumbing for you. Conceptually, all you need to do is find the `MethodDesc*` for the method you want to call, find a managed object for the "this" pointer (if you're calling an instance method), pass in an array of arguments, and deal with the return value. Internally, you'll need to potentially toggle your thread's state to allow the GC to run in preemptive mode, etc.
 
-Here's a simplified example. Note how this instance uses the binder described in the previous section to call SafeHandle's virtual ReleaseHandle method.
+Here's a simplified example. Note how this instance uses the binder described in the previous section to call `SafeHandle`'s virtual `ReleaseHandle` method.
 
-	void SafeHandle::RunReleaseMethod(SafeHandle* psh)
-	{
-	    CONTRACTL {
-	        THROWS;
-	        GC_TRIGGERS;
-	        MODE_COOPERATIVE;
-	    } CONTRACTL_END;
+```C++
+void SafeHandle::RunReleaseMethod(SafeHandle* psh)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    } CONTRACTL_END;
 
-	    SAFEHANDLEREF sh(psh);
+    SAFEHANDLEREF sh(psh);
 
-	    GCPROTECT_BEGIN(sh);
+    GCPROTECT_BEGIN(sh);
 
-	    MethodDescCallSite releaseHandle(s_pReleaseHandleMethod, METHOD__SAFE_HANDLE__RELEASE_HANDLE, (OBJECTREF*)&sh, TypeHandle(), TRUE);
+    MethodDescCallSite releaseHandle(s_pReleaseHandleMethod, METHOD__SAFE_HANDLE__RELEASE_HANDLE, (OBJECTREF*)&sh, TypeHandle(), TRUE);
 
-	    ARG_SLOT releaseArgs[] = { ObjToArgSlot(sh) };
-	    if (!(BOOL)releaseHandle.Call_RetBool(releaseArgs)) {
-	        MDA_TRIGGER_ASSISTANT(ReleaseHandleFailed, ReportViolation)(sh->GetTypeHandle(), sh->m_handle);
-	    }
+    ARG_SLOT releaseArgs[] = { ObjToArgSlot(sh) };
+    if (!(BOOL)releaseHandle.Call_RetBool(releaseArgs)) {
+        MDA_TRIGGER_ASSISTANT(ReleaseHandleFailed, ReportViolation)(sh->GetTypeHandle(), sh->m_handle);
+    }
 
-	    GCPROTECT_END();
-	}
+    GCPROTECT_END();
+}
+```
 
 # Interactions with Other Subsystems
 
 ## Debugger
 
 One limitation of FCalls today is that you cannot easily debug both managed code and FCalls easily in Visual Studio's Interop (or mixed mode) debugging. Setting a breakpoint today in an FCall and debugging with Interop debugging just doesn't work. This most likely won't be fixed.
 
-# Physical Architecture
+# Physical architecture
```

Comment:
I see that you have lower-cases some titles (except the first word), but not all of them. Is there a pattern or is it just an oversight?

Reply:
Probably an oversight. Will go through again.